src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt

```diff
@@ -0,0 +1,59 @@
+package org.rust.lang.refactoring.extractFunction
+
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.ide.utils.findStatementsInRange
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.ext.parentOfType
+import org.rust.lang.core.psi.ext.selfParameter
+
+class RsExtractFunctionConfig(val file: PsiFile, start: Int, end: Int) {
+    var implType = RsWrapperType.Function
+    var anchor: RsFunction? = null
+    var elements: List<PsiElement> = emptyList()
+    var name = ""
+    var visibilityLevelPublic = false
+
+    init {
+        init(start, end)
+    }
+
+    private fun init(start: Int, end: Int) {
```

Comment:
Why do we need this to be a separate function? Can we just move all this into the `init` block?